### ISSUE TITLE
KeyError: 'segmentName' when using smoothing tool in segment editor

### ISSUE THREAD

**Issue by evelyndooley** [Slicer/Slicer#7482]
## Summary

Smoothing tool in segment editor isn't working in my project. I tried 5.5.0, 5.6.1, and 5.7.0-12-16 they all gave the following error:

```
Traceback (most recent call last):
  File "C:/Users/evely/AppData/Local/slicer.org/Slicer 5.7.0-2023-12-16/lib/Slicer-5.7/qt-scripted-modules/SegmentEditorEffects/SegmentEditorSmoothingEffect.py", line 230, in onApply
    self.showStatusMessage(_("Smoothing {segmentName}...").format(segmentationNode.GetSegmentation().GetSegment(segmentID).GetName()))
KeyError: 'segmentName'
```

## Steps to reproduce

1. Select the segmentation
2. Select the source volume
3. Choose the segment
4. Select Smoothing tool
5. Click "Apply" => Above error appears in error log => ERROR: Slicer should apply smoothing instead

## Environment
- Slicer version: Slicer-5.5.0, 5.6.1, and 5.7.0-12-16
- Operating system: Windows 11

**Comment by evelyndooley** [comment]
Here is a screenshot of the interface. All 3 errors are the same from clicking 'Apply'
![image](https://github.com/Slicer/Slicer/assets/31457788/3ded331a-a73e-4a7d-9856-4b65419fe001)

**Comment by evelyndooley** [comment]
Commenting out line 230 in SegmentEditorSmoothingEffect.py seems to allow smoothing to work again. Going to play around with the script some more and open a PR.

**Comment by lassoan** [comment]
Thanks for the detailed error report, it was very useful. To reproduce the issue, I had to enable "Apply to visible segments" option as well. The issue could evade detection due to this, as it is not used very commonly (for smoothing multiple segments it is usually better to use joint smoothing).

Anyway, I'll push an update right away that will make smoothing of all segments work again in Slicer-5.7 releases that are downloaded tomorrow or later.

**Comment by che85** [comment]
It seems, this was not fixed in the most recent stable release.

**Comment by jamesobutler** [comment]
That is correct. The fix is only in latest Slicer 5.7 preview. It was fixed in https://github.com/Slicer/Slicer/commit/2ec2c1de6b8f90f61925a009bdea35179db6012d which was integrated 4 days after Slicer 5.6.1 which was released on December 12th 2023. Slicer 5.6.2 was released on April 5th 2024, but with very minimal bug fixes backported for that release. The fix will be included in the next Slicer stable release which has a date to-be-determined. The most recent discussion was mentioned in https://discourse.slicer.org/t/2024-07-23-weekly-meeting/37505/4.
